### DISCUSSION TITLE
Missing feature: a way to SET (not sync) blend-shapes on base avatar

### DISCUSSION THREAD

**Discussion by Rei-57645** [bdunderscore/modular-avatar#835] [Ideas]
Hello.
For many booth outfits, you _sadly_ have to hide legs / arms etc with certain "shrink blendshapes". E.g. "shrink_leg" + "shring_feet" etc have to be set to 100, to wear pants from a modular outfit. Now the animations from the modular outfit can not access & animate the blend-shapes from the base avatar (makes sense, because we want modularity).

There is already a way to sync blend-shapes (like "boob size) - but the above mentioned (setting a blendshape to a value) is not possible, but often required. A condition for setting the required blend shapes could be the visibility of a mesh renderer (the pants) - or something else.

Also related: how do I hide existing outfits from the base avatar, when I want to show an outfit that was added by MA ? Yes, the MA module can bring its own params and FX layer that gets merged in, but I can not hide outfits existing on the avatar. (no access / visibility). I _could_ use a VRC parameter driver to disable them - but then the avatar is naked when you undress the MA outfit again.

greetings, Rei <3

**Comment by bdunderscore** [comment]
This is an interesting concept to come after the mesh preview feature I’m planning for NDMF - the important thing is that the state of the avatar should match between editor and build time, but we don’t want to destructively change things, so it needs something similar to scale adjuster here…

**Comment by bdunderscore** [comment]
This is now supported as of 1.10!
